### DISCUSSION TITLE
Feature Request: Event trigger on PR review comment resolution change

### DISCUSSION THREAD

**Discussion by melink14** [community/community#27094] [Actions]
<p>I created a workflow which fails if there are unresolved comments on a PR to ensure nothing is forgotten and to make auto-merging easier.</p>
<p>It works as expected for new unresolved comments but I can’t find any event which triggers upon clicking the “resovle” button meaning that unless someone triggers the workflow manually it will be hard to get into the passing state consistently.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/melink14">@melink14</a> ,</p>
<p>Yes, there is not any events or activity types of the <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review_comment" rel="noopener nofollow ugc"><strong>pull_request_review_comment</strong></a> event for resolving the conversation.<br>
I tested and tried, however I did not find any other workarounds.</p>
<p>If your projects really need this feature, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="noopener nofollow ugc">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by melink14** [comment]
<p>Thanks, brightran, for trying to see if there were some workarounds.</p>
<p>I left feedback for this usecase via the feedback form.</p>
<p>I hope as the code review abilities become more mature this type of functionality will be prioritized. :slight_smile:</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/melink14">@melink14</a> ,</p>
<blockquote>
<p>I hope as the code review abilities become more mature this type of functionality will be prioritized.</p>
</blockquote>
<p>Yes, I agree that.</p>
<p>In addition, you also can follow the “<a href="https://github.com/github/roadmap" rel="noopener nofollow ugc"><strong>GitHub public roadmap</strong></a>” repository where you can learn about what features the appropriate engineering teams are working on, what stage the features are in, and when the teams expect to bring them to you.</p>
<p>When the new features are published, you can view the release note from “<a href="https://github.blog/changelog/" rel="noopener nofollow ugc"><strong>GitHub Changelog</strong></a>”.</p>

**Comment by SamuelCabralCruz** [comment]
<p>I created the following <a href="https://github.com/marketplace/actions/unresolved-review-threads" rel="noopener nofollow ugc">GitHub Action</a> to prevent the merge of a pull request with unresolved review threads. The flow is far from perfect, but it’s a start! Hope it is of any help!</p>

**Comment by melink14** [comment]
<p>Thanks for the update <a class="mention" href="/samuelcabralcruz">@SamuelCabralCruz</a>. I may try it out. I’ve just been working on this as well using composite actions but I like the idea of using a label to work around the lack of event for comment resolution.</p>
<p>One thing to note is that your action may fail for PRs from external forks since GitHub recently restricted those workflows to read only.</p>
<p>In the case of adding labels we can use <code>pull_request_target</code> instead but that may not work for things like <code>pull_request_review_comment</code> or <code>pull_request_review</code>.</p>
<p>Though the docs are not clear on this so it will take some experimentation.</p>

**Comment by melink14** [comment, marked as answer]
<p>There’s a new branch protection in beta which seems to do what I need though I haven’t tested it extensively.<br>
<a href="https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-conversation-resolution-before-merging" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-conversation-resolution-before-merging</a></p>
<p>I imagine there could be other reasons to have a resolution event trigger for workflows but this seems like it will meet the needs in this thread better than we can do ourselves!</p>
